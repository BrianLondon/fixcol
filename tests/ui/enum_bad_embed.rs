use fixcol_derive::ReadFixed;

#[derive(ReadFixed)]
struct MyStruct {
    #[fixcol(width = 5)]
    my_int: u16
}

#[derive(ReadFixed)]
#[fixcol(key_width = 3)]
enum Thing {
    #[fixcol(key = "one", embed = true)]
    Thing1(#[fixcol(width = 5)] u16),
    #[fixcol(key = "two", embed = true)]
    Thing2(MyStruct),
}

pub fn main() {}
