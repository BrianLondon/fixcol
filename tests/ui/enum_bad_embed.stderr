error: fixed-derive error: Did not expect fixed attribute on embedded enum variant
  --> tests/ui/enum_bad_embed.rs:13:14
   |
13 |     Thing1(#[fixed(width = 5)] u16),
   |              ^^^^^
